.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,361 @@
+          DATABASE_USER: mlp
+          DATABASE_PASSWORD: mlp
+          KETO_URL: http://localhost:4466
+        run: make it-test-api-ci
+
+      - name: Lint code
+        run: |
+          make setup
+          make lint-api
+          
+
+  e2e-test:
+    runs-on: ubuntu-latest
+    needs:
+      - build-api
+      - test-api
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.9.3
+
+      - name: Setup local k8s cluster
+        uses: AbsaOSS/k3d-action@v2.1.0
+        with:
+          cluster-name: mlp-e2e
+          args: >-
+            --agents 1
+            --port 80:80@agent:*
+            --registry-create ${{ env.LOCAL_REGISTRY }}
+            --k3s-arg "--no-deploy=metrics-server@server:*"
+
+      - name: Download API image tar
+        uses: actions/download-artifact@v3
+        with:
+          name: mlp-api.${{ needs.build-api.outputs.api-version }}.tar
+
+      - name: Publish image to the local registry
+        env:
+          DOCKER_REPOSITORY: ${{ env.LOCAL_REGISTRY }}/${{ github.repository }}
+        shell: bash
+        run: |
+          docker image load --input mlp-api.${{ needs.build-api.outputs.api-version }}.tar
+          docker tag \
+            mlp-api:${{ needs.build-api.outputs.api-version }} \
+            ${{ env.DOCKER_REPOSITORY }}/mlp-api:${{ needs.build-api.outputs.api-version }}
+          docker push ${{ env.DOCKER_REPOSITORY }}/mlp-api:${{ needs.build-api.outputs.api-version }}
+
+      - name: Deploy MLP to the local cluster
+        run: |
+          helm repo add caraml https://caraml-dev.github.io/helm-charts/
+          
+          kubectl create ns mlp
+          
+          helm install mlp caraml/mlp --namespace=mlp \
+            --wait --timeout=5m \
+            --values=.github/e2e/values.yaml \
+            --set mlp.image.registry=${{ env.LOCAL_REGISTRY }} \
+            --set mlp.image.repository=${{ github.repository }}/mlp-api \
+            --set mlp.image.tag=${{ needs.build-api.outputs.api-version }}
+          
+          echo "::group::Check cluster resources"
+          kubectl get all --namespace=mlp
+          kubectl get ingress --namespace=mlp
+          echo "::endgroup::"
+
+      - name: Setup MLP project
+        shell: bash
+        run: |
+          sleep 2
+          
+          tee payload.json <<EOF
+          {
+            "name": "e2e-test",
+            "team": "myteam",
+            "stream": "mystream"
+          }
+          EOF
+
+          curl -v \
+            --header 'Content-Type: application/json' \
+            --request POST \
+            --data @payload.json \
+            http://mlp.127.0.0.1.nip.io/v1/projects
+
+      - name: Run E2E tests
+        env:
+          MLP_API_BASEPATH: http://mlp.127.0.0.1.nip.io/v1
+        shell: bash
+        run: |
```

Comment:
Is there any value in turning this into a make target?

Reply:
To be honest, this isn't a proper e2e test, so I decided to keep it as is for the time being.